### DISCUSSION TITLE
How to use new model parameters file *.pars

### DISCUSSION THREAD

**Discussion by SiWang4phys** [FeynCalc/feyncalc#243] [Q&A]
Hi,

I generated a new model using FeynRules, it has three model files: *.gen, *.mod, *.pars.
Where FAPatch patched 2 models, the parameters did not get the values.
Here is part of the content in the *.pars file:
`(*             External Parameters             *)
(***********************************************)

M$ExtParams = {
        aEWM1 -> 127.9,
        aS -> 0.1184,
        cabi -> 0.227736,
        cdBAbs1x1 -> 1,
        cdBAbs1x2 -> 1,
        cdBAbs1x3 -> 1,
        cdBAbs2x1 -> 1,
        cdBAbs2x2 -> 1,
        cdBAbs2x3 -> 1,
        cdBAbs3x1 -> 1,
        cdBAbs3x2 -> 1,
        cdBAbs3x3 -> 1,
        cdBPh1x1 -> 0,
        cdBPh1x2 -> 0,
        cdBPh1x3 -> 0,
        cdBPh2x1 -> 0,
        cdBPh2x2 -> 0,
        cdBPh2x3 -> 0,
        cdBPh3x1 -> 0,
        cdBPh3x2 -> 0,
`
How do I use this pars file? How to replace all the numerical values for the variables?

Thanks,
Si

**Comment by vsht** [comment]
Patching just renames some symbols, it doesn't do anything with the model itself and it doesn't apply replacements.

I don't think that FeynArts applies the parameters automatically at all. So you need to do it by hand.

**Comment by SiWang4phys** [comment]
Thanks! That explains why I cannot find any examples.
